src/extension.ts

```diff
@@ -85,6 +89,18 @@ export async function activateInternal(ctx: vscode.ExtensionContext, perfStats:
             )
         );
 
+        if (ext.outputChannel.debugLoggingEnabled) {
```

Comment:
This has me wondering. You mentioned VSCode doesn't really expose anything about this channel until it is registered. If the log level is changed _after_ activation, all of this code will be missed. I'm fine with that if the log level change persists to the next session, but if not then none of this code could ever run.

Reply:
There is a hook to detect log level changes, but it doesn't offer any way of knowing what the previous log level was. We could potentially add a check to see if we've logged these diagnostics before and only show them once, but I think for the handful of startup properties it's reasonable to ask users to enable logging then restart VS Code in order to capture the full debug context.

Reply:
Right. I'm totally fine requiring a restart to get the full output. I just don't know if VSCode "forgets" inbetween sessions that an extension has had its logging level bumped up.